iceoryx_hoofs/include/iceoryx_hoofs/cxx/unique_ptr.hpp

```diff
@@ -91,28 +84,50 @@ class unique_ptr
 
     ///
     /// @brief release Release ownership of the underlying pointer.
-    /// @return Pointer to the managed object or nullptr if none owned.
+    /// @param[in] ptrToBeReleased unique_ptr which is destroyed without deleting its underlying data
+    /// @return Pointer to the managed object.
     ///
-    T* release() noexcept;
+    static T* release(unique_ptr&& ptrToBeReleased) noexcept;
 
     ///
     /// @brief reset Reset the unique pointer to take ownership of the given pointer.
-    /// @details Any previously owned objects will be deleted. If no pointer given then points to nullptr.
-    /// @param ptr Pointer to object to take ownership on.
+    /// @details Any previously owned objects will be deleted.
+    /// @param ptr Pointer to object to take ownership on. It is forbidden to provide the value nullptr!
```

Comment:
You could think about adding a hint, that the application might terminate in debug mode when a `nullptr` is provided.

Reply:
I would say also in release mode it should terminate. Otherwise it segfaults as soon as one calls `operator->` and then we should terminate as soon as possible

Reply:
@elfenpiff After a discussion with @MatthiasKillat we came to the conclusion that we would prefer the "traditional":

* Release mode: all assert / Expects / Ensures are no-ops for performance reason e.g. out of bounds check in `cxx::vector`
* Debug mode: all assert / Expects / Ensures are active

It's recommend for iceoryx user to test their software in debug mode extensively and only then switch to the release mode.

What do you think?

Reply:
I think this discussion is not part of this pull request. But to clarify, I call `Ensures(ptr != nullptr)` in the implementation and this can be replaced with anything which we come up with in the non recoverable error handling design.

I think `IOX_ASSERT` would be its successor and at this point I do not care about its behavior.

Reply:
@elfenpiff Ok, makes sense. 

@MatthiasKillat Friendly reminder that we might to update few doxygen places e.g. here or in `cxx::optional` where it states `Will terminate if empty object is accessed`.